### ISSUE TITLE
Aliases of views such as (SELECT a AS a1, a AS a2...) become mixed up

### ISSUE THREAD

**Issue by julianmenzler** [hyrise/hyrise#1768]
Since #1692, the SQLTranslator supports queries like `SELECT a AS a1, a AS a2...`. 

However, there is a bug when building LQPs from views in the StorageManager or WITH clauses. 

I created the following test to demonstrate the problem:
```cpp
  // Define and create view
  const auto view_node =
      compile_query("SELECT a AS a1, a AS a2 FROM int_float WHERE a > 10");
  const auto view_columns = std::unordered_map<ColumnID, std::string>({
                                                                          {ColumnID{0}, "a1"},
                                                                          {ColumnID{1}, "a2"}
                                                                      });
  const auto view = std::make_shared<LQPView>(view_node, view_columns);
  StorageManager::get().add_view("alias_view", view);

  // Use the view within a query
  const auto actual_lqp = compile_query("SELECT * FROM alias_view");

  const auto aliases = std::vector<std::string>({"a1", "a2"});
  // Redundant AliasNode due to the SQLTranslator architecture. Doesn't look nice, but not really an issue.
  const auto expected_lqp =
    AliasNode::make(expression_vector(int_float_a, int_float_a), aliases,
      AliasNode::make(expression_vector(int_float_a, int_float_a), aliases,
                      ProjectionNode::make(expression_vector(int_float_a, int_float_a),
                                           PredicateNode::make(greater_than_(int_float_a, value_(10)),
                                                               stored_table_node_int_float))));

  EXPECT_LQP_EQ(actual_lqp, expected_lqp);
```

Here is the output:
```
-------------- Actual LQP --------------
[0] [Alias] a AS a2, a AS a2
 \_[1] [Alias] a AS a1, a AS a2
    \_[2] [Projection] a, a
       \_[3] [Predicate] a > 10
          \_[4] [StoredTable] Name: 'int_float' pruned: 0/1 chunk(s), 0/2 column(s)

------------- Expected LQP -------------
[0] [Alias] a AS a1, a AS a2
 \_[1] [Alias] a AS a1, a AS a2
    \_[2] [Projection] a, a
       \_[3] [Predicate] a > 10
          \_[4] [StoredTable] Name: 'int_float' pruned: 0/1 chunk(s), 0/2 column(s)
```

As you can see, `a1` becomes `a2`.

The root of the problem can be found in `SQLTranslator::_translate_table_origin`: 
https://github.com/hyrise/hyrise/blob/691f33baea58a30a0b254e87b6fc37db7c42d60b/src/lib/sql/sql_translator.cpp#L497-L500

Here, the SQLIdentifierResolver maps expressions to identifiers. The scope of aliases, limited to individual select-list-elements, is ignored.

**Comment by mrks** [comment]
Another one that might be related: `SELECT x+1 FROM (SELECT a AS x FROM t) t2` returns a column `a+1`, not `x+1`. Please open a new issue if this is unrelated.

**Comment by mrzzzrm** [comment]
Yes, this behaviour is to be expected with the way expression-to-expression (i.e. column-to-column) mapping is handled in LQPs. As previously stated, when designing this, I did not consider cases where a) a node emits the same column multiple times or b) where multiple input nodes contain what looks like the same column by identity (think `count(*)`, `random()`) but are actually different columns.

It get's worse. Consider these two LQPs:

![photo_2019-08-25_16-21-42](https://user-images.githubusercontent.com/2679018/63651398-87bcfa00-c754-11e9-9e02-4bba14d68e87.jpg)

Both have a Projection at the top that has ambiguous column references that it would right now silently ignore - and these are *not* "non-column" expressions like `count(*)` and `random()` are!

So, I think this a rather systemic problem, and not something that will be easy to fix - an easy fix will likely break on the next slightly different query. I could not come up with a solution yet, sorry. Will report back if I can think of something.

**Comment by julianmenzler** [comment]
I have to admit, very ugly LQPs-scenarios. Thanks for your feedback @mrzzzrm!

**Comment by janetzki** [comment]
We could solve at least the example test if we remove the secondary AliasNodes as described in #1186.

**Comment by mrks** [comment]
https://github.com/hyrise/hyrise/blob/da63eecf05e9793164c212cdc7ce716a01d4054a/src/lib/sql/sql_translator.cpp#L276

Might be fixable if we either correctly fill `element.identifiers` when using `WITH`, a view, or a subselect or, in case that `element.identifiers.empty()`, traverse the LQP and check for a previous AliasNode.
